admiral/pkg/apis/admiral/model/dependencyproxy.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+package admiral.global.v1alpha;
+
+option go_package = "model";
+
+// The below example of DependencyProxy
+//```yaml
+//    apiVersion: admiral.io/v1alpha1
+//    kind: DependencyProxy
+//    metadata:
+//      name: dependency-proxy-example
+//      namespace: admiral
+//      annotations:
+//          admiral.io/env: stage
+//    spec:
+//      destination:
+//          identity: greeting
+//          dns_suffix: "xyz"
+//          dns_prefix:
+//              - "test0"
+//              - "test1"
+//      proxy:
+//          identity: nginx-gw
+//```
+// The above DependencyProxy will generate the following
+// VirtualService object
+//```yaml
+//  apiVersion: networking.istio.io/v1alpha3
+//  kind: VirtualService
+//  metadata:
+//      name: httpbin-vs
+//  spec:
+//      hosts:
+//       - test0.stage.greeting.xyz
+//       - test1.stage.greeting.xyz
```

Comment:
I am thinking if these should be test0.greeting.xyz and test1.greeting.xyz because if some one has e2e1 and e2e2, the endpoints will look like e2e1.e2e.greeting.xyz and e2e2.e2e.greeting.xyz. Also this will make the migration later (from gateway to mesh seamless when that happens)

Reply:
hey @aattuluri , I'm sorry this example I think is a little confusing. `stage` here is the `admiral.io/env` on the deployment/rollout of the destination service(greeting). So I was thinking if we'll prepend the prefixes defined above to the generated endpoint of the service. `<env>.<identity>.<suffix>

So we'll be generating the regular endpoint without the prefix as well as shown here https://github.com/istio-ecosystem/admiral/blob/02fd5ef60ab3e3996addb37f1408e9a67aa9a5ec/admiral/pkg/apis/admiral/model/dependencyproxy.proto#L37
This should make the migration seamless. Please correct me if I'm wrong.

Reply:
Makes sense.